src/process_utility.c

```diff
@@ -1114,6 +1114,46 @@ process_truncate(ProcessUtilityArgs *args)
 							 */
 							list_changed = true;
 					}
+					else if ((chunk = ts_chunk_get_by_relid(relid, false)) != NULL)
+					{
+						ht = ts_hypertable_cache_get_entry(hcache,
+														   chunk->hypertable_relid,
+														   CACHE_FLAG_NONE);
+
+						Assert(ht != NULL);
+
+						/* If the hypertable has continuous aggregates, then invalidate
+						 * the truncated region. */
+						if (ts_continuous_agg_hypertable_status(ht->fd.id) == HypertableIsRawTable)
+						{
+							int64 start = ts_chunk_primary_dimension_start(chunk);
+							int64 end = ts_chunk_primary_dimension_end(chunk);
+
+							Assert(hyperspace_get_open_dimension(ht->space, 0)->fd.id ==
+								   chunk->cube->slices[0]->fd.dimension_id);
+							ts_cm_functions->continuous_agg_invalidate_raw_ht(ht, start, end);
```

Comment:
This is done elsewhere and would probably benefit from having a single function created for this job. Something like `ts_cagg_invalidate_chunk(chunk)`.